pep-0570.rst

```diff
@@ -76,7 +78,8 @@ the developer to maintain its name forever, as users might pass ``x`` as a
 keyword.
 
 Another good example is an API that wants to transmit the feeling
-of ownership through positional arguments, see::
+that one of its parameters is the "main" parameter through positional
+arguments. For example, see ::
 
     class MyDecorator:
         def __init__(self, original_function):
```

Comment:
note for myself, I didn't know that self can be passed as a keyword argument :-)

```
class A:
    def meth(self, arg):
        print(self, arg)

A.meth(self="hello", arg="arg")
```

Output:

```
hello arg
```

I'm not sure that it's a good API than self=value is allowed :-)

Reply:
Good point!

Reply:
More fun with self.

```
>>> import io
>>> help(io.FileIO.write)
Help on method_descriptor:

write(self, b, /)
    Write buffer b to file, return number of bytes written.

>>> f=io.FileIO("/dev/null", "wb")
>>> io.FileIO.write(f, b"data")  # ok
4

>>> io.FileIO.write(self=f, b=b"data") # wtf?
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: descriptor 'write' of '_io.FileIO' object needs an argument
```

It seems like 'method_descriptor' handles self as a positional-only argument since (at least) Python 2.7, but the error message isn't helpful. I created: https://bugs.python.org/issue36437